ddapm_test_agent/checks.py

```diff
@@ -85,6 +109,31 @@ def frames_dfs(self) -> Generator[Tuple[CheckTraceFrame, int], None, None]:
     def has_fails(self) -> bool:
         return len([f for f in self.frames() if f.has_fails()]) > 0
 
+    def get_results(self, results: Dict[str, Dict[str, int]]) -> Dict[str, Dict[str, int]]:
+        for f in self.frames():
+            results = f.get_results(results)
```

Comment:
do you mean to overwrite results each time?

Reply:
yes, we want the trace check results to be updated after each request, which means looping through each frame to get the results from that specific frame.

Reply:
oh, I think I see.  

What you probably want here is `update_results(...)` .  Since dicts are passed by reference in Python, the results dict will automatically be updated.

Reply:
fixed